docs/tpm2-pkcs11_object_auth_model.md

```diff
@@ -0,0 +1,528 @@
+attributes - userwithauth and adminwithpolicy.<br>
+<br>
+It is also possible to associate the authorization of one object as the
+authorization of another object. One caveat of changing authorization value of
+a secondary object is that it does not invalidate the old object, instead
+creates a new object that protects the same sensitive information. Also, while
+auth values can be altered the authorization policy cannot be changed.
+
+### The NV Index/ Object
+There are two types of TPM NV objects - those that are made persistent after
+creating/ importing the objects under a TPM hierarchy OR those defined as NV
+indexes to hold user defined data. One can define an NV index of size 0 with an
+authorization password and or policy for administrative or user operations.
+Unlike the secondary objects, the NV index object authorization is not used in
+the encryption of it's contents - the NV index object authorization simply
+protects read and write access to the object. Thus no new NV indexes need to be
+defined when changing the authorization.
+
+## tpm2-pkcs11 objects
+A general overview of the object and authorization model design with TPM2
+objects. The SO and USER roles separation is realized in the tpm2-pkcs11 design
+by two unique TPM2 objects, either transient objects or NV indexes. The
+authorization values chosen for these objects are further utilized in creating
+specialized authorization policies that render the special properties of the
+SO-PIN, USER-PIN, Wrapping-Key-Sealing and the USER(keys) objects.
+1. SO-PIN object: The auth value of this object authorizes auth value change of
+SO-PIN and USER-PIN objects. This is either an NV index or a transient secondary
+object.
+2. USER-PIN object: The auth value of this object is used as an auth value of
+USER-PIN, Wrapping-Key-Sealing, and USER(keys) objects. The USER-PIN object auth
+value OR SO-PIN object auth value can authorize change of USER-PIN object auth
+value. This is either an NV index or a transient secondary object.
+3. Wrapping-Key-Sealing object: The authorization value of this object is
+referenced off the USER-PIN object with a policy. The object itself doesn't have
+a auth value of it's own and so the unsealing user operation would need the auth
+policy requiring USER pin ought to be satisfied. It seals a wrapping key used to
+encrypt the auth value of existing TPM2 objects to be assumed into the
+tpm2-pkcs11 model. This is a transient secondary object.
+4. USER(keys) object: The authorization model is same as the
+Wrapping-Key-Sealing object and thus requires the USER-PIN object auth value via
+a policy. These are user created keys and objects that are used in user
+operations like encryption, decryption, signing, verification, etc. These are
+secondary transient objects.
+
+## NV object diagram
+Overview of the use of the TPM2 NV indexes in the implementation.
+<p>
+  <img src="illustrations/object_diagram_nv_objects.png"
```

Comment:
Existing TPM objects are those that are already created without using the tpm2-pkcs11 but ought to be used as pkcs11 objects.

Reply:
Pretty much all existing objects will have a password based auth... thats still the predominant auth model.

Reply:
I see. Previous to this point, I don't think the doc mentioned as a design consideration that it wants to support importing pre-existing TPM objects. I think that's worth stating explicitly, along with the security considerations (eg, since an auth value is just being wrapped, an attacker who gets the auth value can continue using it even after PIN changes and even in the NV model).

Reply:
The document should explicitly call that out, and I don't think it is as you point out. That would be helpful to the reader.

The Secondary Object paragraph mentions the object being changeable:
```
It is also possible to associate the authorization of one object as the authorization of another object. One caveat of changing authorization value of a secondary object is that it does not invalidate the old object, instead creates a new object that protects the same sensitive information. Also, while auth values can be altered the authorization policy cannot be changed.
```
But I think where it mentions the fallback model, it would be good to re-iterate that point there.